File: PythonPerformanceTest/INIPerformanceTest.py
import configparser


class IniEntry:
    def __init__(self, SECTION: str, KEY: str, VALUE: str):
        # Not possible to make private variables. possible security issue?
        self.SECTION = SECTION
        self.KEY = KEY
        self.VALUE = VALUE


def execute():
    ini_entries = []
    try:
        config = configparser.ConfigParser()
        config.read("performance_test.ini")
        for section in config.sections():
            for i in range(1, 6):
                ini_entries.append(IniEntry(section, section + str(i), config[section][section + str(i)]))
            last_entry = ini_entries[-1]
            config[last_entry.SECTION][last_entry.KEY] = "Testing around"
            config[last_entry.SECTION][last_entry.KEY] = last_entry.VALUE
    except Exception as e:
        print(e)

File: Old performance Test directories/CSharpPerformanceTest/Ini_Performance_Test.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using IniParser;
namespace CSharpPerformanceTest
{
    public class IniPerformanceTest
    {
        class IniEntry
        {
            public string SECTION { get; set; }
            public string KEY { get; set; }
            public string VALUE { get; set; }
            public IniEntry(string SECTION, string KEY, string VALUE)
            {
                this.KEY = KEY;
                this.SECTION = SECTION;
                this.VALUE = VALUE;
            }
        }
        public void Execute()
        {
            var iniEntries = new List<IniEntry>();
            try
            {
                var parser = new FileIniDataParser();
                var ini = parser.ReadFile("performance_test.ini", Encoding.UTF8);
                var sections = ini.Sections;
                foreach (var sectionName in sections)
                {
                    for (int i = 0; i < 6; i++)
                    {
                        iniEntries.Add(new IniEntry(sectionName.SectionName, sectionName.SectionName + i, ini[sectionName.SectionName][sectionName.SectionName + i]));
                    }
                    var lastEntry = iniEntries.Last();
                    ini[lastEntry.SECTION][lastEntry.KEY] = "Testing around";
                    parser.WriteFile("performance_test.ini", ini, Encoding.UTF8);
                    ini[lastEntry.SECTION][lastEntry.KEY] = lastEntry.VALUE;
                    parser.WriteFile("performance_test.ini", ini, Encoding.UTF8);
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }
        }
    }
}